### PR TITLE
fix: added execution support of onResponse and onError in API block d…

### LINKED ISSUE #465: onResponse of the apis under API block are not working
**Describe the bug**
`onResponse` of the apis under API block of the screen are not working.

**To Reproduce**
Checkout this yaml
```yaml
View:
  onLoad:
    executeCode: |
      ensemble.invokeAPI('getDummyProducts');
  body:
    Text:
      text: This is a test

API:
  getDummyProducts:
    method: GET
    uri: https://dummyjson.com/products
    onResponse:
      executeCode: |
        console.log('response');
```

**Expected behavior**
`onResponse` working

### PR DESCRIPTION
**Pull request by sagardspeed2**
Added support for execution of onResponse and onError block of API block definition in sequential with invokeAPI's onResponse and onError

## Issue ticket number and link

Closes #465 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/hooks/useEnsembleAction.tsx [hunk 1]
```diff
@@ -205,18 +205,20 @@ export const useInvokeAPI: EnsembleActionHook<InvokeAPIAction> = (action) => {
...
   const api = useMemo(
     () => apis?.find((model) => model.name === action?.name),
     [action?.name, apis],
   );
 
+  const onInvokeAPIResponseAction = useEnsembleAction(action?.onResponse);
+  const onInvokeAPIErrorAction = useEnsembleAction(action?.onError);
+
+  const onAPIResponseAction = useEnsembleAction(api?.onResponse);
+  const onAPIErrorAction = useEnsembleAction(api?.onError);
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: Don't call hook multiple times per callback. Just get the correct callback model once.

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/runtime/hooks/useEnsembleAction.tsx (hunk 1 above)
Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: Discussed offline, both are needed if there are multiple onResponse for onLoad and on the API definition. However we still need to handle this for the code `invokeAPI` case.

Comment (sagardspeed2) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: This applies to invokeAPI too. The API is only called when you use invokeAPI. Even during onLoad, invokeAPI is used, triggering the API in that case.

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: I meant when you call ensemble.invokeAPI from code? It works there too?

Comment (sagardspeed2) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: No, in `ensemble.invokeAPI` we are not passing any onResponse or onError callback. so it is not intended to work there

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: Right, but when you call `ensemble.invokeAPI`, should the `onResponse` of the API model fire?

Comment (sagardspeed2) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: No

Comment (anserwaseem) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:220]: > Right, but when you call `ensemble.invokeAPI`, should the `onResponse` of the API model fire?

It should

Comment (evshi) [reply packages/runtime/src/runtime/hooks/useEnsembleAction.tsx:222]: I'll add this in a separate PR

**Review by evshi — APPROVED** [review]

**Pull request merged**